<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>930</width>
    <height>540</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>275</width>
    <height>390</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>930</width>
    <height>540</height>
   </size>
  </property>
  <property name="font">
   <font>
    <stylestrategy>PreferDefault</stylestrategy>
   </font>
  </property>
  <property name="windowTitle">
   <string>3D SCANNER - CSE395 - GROUP 7</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>80</y>
      <width>241</width>
      <height>311</height>
     </rect>
    </property>
    <property name="title">
     <string>Clipping Options</string>
    </property>
    <widget class="QWidget" name="formLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>221</width>
       <height>269</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <property name="labelAlignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
      <property name="formAlignment">
       <set>Qt::AlignHCenter|Qt::AlignTop</set>
      </property>
      <item row="1" column="0">
       <widget class="QLabel" name="xMinLabel">
        <property name="text">
         <string>xMin</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="xMinDoubleSpinBox">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="minimum">
         <double>-99.000000000000000</double>
        </property>
        <property name="maximum">
         <double>99.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="xMaxLabel">
        <property name="text">
         <string>xMax</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="xMaxDoubleSpinBox">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="minimum">
         <double>-99.000000000000000</double>
        </property>
        <property name="maximum">
         <double>99.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="yMinLabel">
        <property name="text">
         <string>yMin</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="yMinDoubleSpinBox">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="minimum">
         <double>-99.000000000000000</double>
        </property>
        <property name="maximum">
         <double>99.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="yMaxLabel">
        <property name="text">
         <string>yMax</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="yMaxDoubleSpinBox">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="minimum">
         <double>-99.000000000000000</double>
        </property>
        <property name="maximum">
         <double>99.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="zMinLabel">
        <property name="text">
         <string>zMin</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QDoubleSpinBox" name="zMinDoubleSpinBox">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="minimum">
         <double>-99.000000000000000</double>
        </property>
        <property name="maximum">
         <double>99.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="zMaxLabel">
        <property name="text">
         <string>zMax</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QDoubleSpinBox" name="zMaxDoubleSpinBox">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="minimum">
         <double>-99.000000000000000</double>
        </property>
        <property name="maximum">
         <double>99.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="captureCountLabel">
        <property name="text">
         <string># Capture </string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="captureCountSpinBox">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="maximum">
         <number>150</number>
        </property>
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="refreshButton">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>20</y>
      <width>44</width>
      <height>44</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>New Scan</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/icons/resources/refresh.png</normaloff>:/icons/resources/refresh.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>24</width>
      <height>24</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="kinectStatusIcon">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>79</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap resource="resources.qrc">:/icons/resources/kinect-disconnected.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>false</bool>
    </property>
    <property name="wordWrap">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="captureButton">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>20</y>
      <width>44</width>
      <height>44</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Start Scanning</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/icons/resources/start.png</normaloff>:/icons/resources/start.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>24</width>
      <height>24</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>470</y>
      <width>241</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>5</number>
    </property>
    <property name="format">
     <string>%p%</string>
    </property>
   </widget>
   <widget class="QVTKWidget" name="vtkWidget">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>10</y>
      <width>640</width>
      <height>480</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(0, 0, 0);</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="cloud_mesh_check_box">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>390</y>
      <width>111</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Cloud/Mesh</string>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QVTKWidget</class>
   <extends>QWidget</extends>
   <header>QVTKWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>xMinDoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setXMin(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xMaxDoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setXMax(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yMinDoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setYMin(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yMaxDoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setYMax(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zMinDoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setZMin(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>185</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zMaxDoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setZMax(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>captureButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onCaptureClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onRefreshClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>730</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateKinectPixmap(QPixmap)</signal>
   <receiver>kinectStatusIcon</receiver>
   <slot>setPixmap(QPixmap)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>479</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel">
     <x>719</x>
     <y>56</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>destroyed()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopScanner()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>479</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>captureCountSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setCaptureCount(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>172</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>137</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cloud_mesh_check_box</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>showCloudMesh(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>464</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>updateKinectPixmap(QPixmap)</signal>
  <slot>setXMin(double)</slot>
  <slot>setYMin(double)</slot>
  <slot>setZMin(double)</slot>
  <slot>setXMax(double)</slot>
  <slot>setYMax(double)</slot>
  <slot>setZMax(double)</slot>
  <slot>onCaptureClick()</slot>
  <slot>onRefreshClick()</slot>
  <slot>onProgressUpdate(int,string)</slot>
  <slot>setCaptureCount(int)</slot>
  <slot>showCloudMesh(bool)</slot>
 </slots>
</ui>
